daemon/logger/loggerutils/tag_test.go

```diff
@@ -42,8 +52,44 @@ func TestGenerateLogTag(t *testing.T) {
 			expected: "",
 			hasError: true,
 		},
+
+		{
+			tag:      "{{with .ExtraAttributes nil}}this line should not show up{{end}}",
+			expected: "",
+			hasError: false,
+		},
+
+		// use labels config
+		{
+			tag:      "{{with .ExtraAttributes nil}}label from is {{.from}}{{end}}",
+			labels:   "from,to",
+			expected: "label from is open-source",
+			hasError: false,
+		},
+
+		// use env or env-regex config
+		{
+			tag:      "{{with .ExtraAttributes nil}}APP is {{.APP}}{{end}}",
+			env:      "APP",
+			expected: "APP is pouchcontainer",
+			hasError: false,
+		}, {
+			tag:      "{{with .ExtraAttributes nil}}APP is {{.APP}}{{end}}",
+			env:      "VERSION",
+			expected: "APP is <no value>",
+			hasError: false,
+		}, {
+			tag:      "{{with .ExtraAttributes nil}}Version is {{.VERSION}}{{end}}",
```

Comment:
how about some  {{with .ExtraAttributes notnil}} test?

Reply:
Since the go template doesn't support the function as args, I add other for the `ExtractAttributes`. 

BTW, the definition of `ExtracAttributes` function which has the `keyMod` is used to compatible with `moby`.

Please take a look